src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1675,15 +1675,27 @@ def _resolve_hostname_through_dns(hostname):
     return socket.gethostbyname(hostname)
 
 
-def create_webapp_slot(cmd, resource_group_name, webapp, slot, configuration_source=None):
+def create_webapp_slot(cmd, resource_group_name, webapp, slot, configuration_source=None,
+                       deployment_container_image_name=None, docker_registry_server_password=None,
+                       docker_registry_server_user=None):
+    container_args = deployment_container_image_name or docker_registry_server_password or docker_registry_server_user
+    # TODO does this constraint make sense?
+    if container_args and not configuration_source:
```

Comment:
@seligj95 as I implemented it, you need to provide a `--configuration-source` param (the slot to copy configs from) in order to override any container settings (`deployment_container_image_name`, `docker_registry_server_password`, and `docker_registry_server_user`). 
Does that constraint make sense or should users be able to set these docker settings without setting `--configuration-source`?

Reply:
Here's what I think it should look like - user needs to specify a configuration source. If user specifies any of the container settings (deployment_container_image_name, docker_registry_server_password, or docker_registry_server_user), use that instead of what is coming from the configuration source. is that how you implemented it?

Reply:
yep that's how I implemented it. Sounds good thanks